plugins/apitracing/test/Extractor_Unittest.cpp

```diff
@@ -0,0 +1,321 @@
+
+            if (addressWidth == ConstantDefinitions::x64AddressWidth)
+            {
+                SetupX64StackReads(parameters);
+            }
+            else
+            {
+                SetupX86StackReads(parameters);
+            }
+            return result;
+        }
+
+        void SetupX64StackReads(std::vector<TestParameterInformation> parameters)
+        {
+            auto stackEntrySize = ConstantDefinitions::x64AddressWidth / ConstantDefinitions::byteSize;
+            auto stackOffset = 0;
+
+            for (size_t i = ConstantDefinitions::maxRegisterParameterCount; i < parameters.size(); i++)
+            {
+                ON_CALL(*introspectionAPI,
+                        read64VA(testRsp + ConstantDefinitions::stackParameterOffsetX64 + stackOffset, testDtb))
+                    .WillByDefault(Return(parameters[i].expectedValue));
+                stackOffset += stackEntrySize;
+            }
+        }
+
+        void SetupX86StackReads(std::vector<TestParameterInformation> parameters)
+        {
+            for (size_t i = 0; i < parameters.size(); i++)
+            {
+                ON_CALL(*introspectionAPI,
+                        read64VA(testRsp + (i + 1) * ConstantDefinitions::stackParameterOffsetX86, testDtb))
+                    .WillByDefault(Return(parameters[i].expectedValue));
+            }
+        }
+
+        static std::vector<ExtractedParameterInformation> SetupExpectedNestedParameters()
+        {
+            auto dummyExtract = std::string("extract me");
+            auto nestedStruct = ExtractedParameterInformation(
+                {.name = "ObjectAttributesOne",
+                 .data = {},
+                 .backingParameters = std::vector<ExtractedParameterInformation>{ExtractedParameterInformation{
+                     .name = "ObjectAttributesTwo",
+                     .data = {},
+                     .backingParameters = std::vector<ExtractedParameterInformation>{ExtractedParameterInformation{
+                         .name = "ObjectAttributesTwoContent", .data = dummyExtract, .backingParameters = {}}}}}});
+
+            return std::vector<ExtractedParameterInformation>{
+                ExtractedParameterInformation{
+                    .name = "FileHandle", .data = static_cast<uint64_t>(param1Value), .backingParameters = {}},
+                ExtractedParameterInformation{nestedStruct}};
+        }
+
+        static std::vector<uint64_t> GetExpectedValues(const std::vector<TestParameterInformation>& testParameters)
+        {
+            std::vector<uint64_t> expectedExtractedParameters{};
+            for (const auto& testParameter : testParameters)
+            {
+                expectedExtractedParameters.emplace_back(testParameter.expectedValue);
+            }
+            return expectedExtractedParameters;
+        }
+
+        static bool variantsAreEqual(const ExtractedParameterInformation& first, // NOLINT(misc-no-recursion)
```

Comment:
The purpose for all those NOLINT comments was to surpress certain lints for the CI, but we've never executed the static analysis on tests.

Reply:
Not quite
It is used in the IDE so suppressing this allows us to find "real" problems more easily.

Reply:
In this case I would consider this personal settings. They should not be code annotations.